packages/app/src/vis-packs/core/utils.ts

```diff
@@ -93,11 +93,33 @@ export function getImageInteractions(keepRatio: boolean): InteractionInfo[] {
   return keepRatio ? BASE_INTERACTIONS : INTERACTIONS_WITH_AXIAL_ZOOM;
 }
 
+function isBigIntArray(val: ArrayValue<NumericLikeType>): val is bigint[] {
+  return Array.isArray(val) && typeof val[0] === 'bigint';
+}
+
 function isBoolArray(val: ArrayValue<NumericLikeType>): val is boolean[] {
   return Array.isArray(val) && typeof val[0] === 'boolean';
 }
 
-export function toNumArray(arr: ArrayValue<NumericLikeType>): NumArray {
+export function toNumArray<T extends ArrayValue<NumericLikeType> | undefined>(
+  arr: T,
+): T extends ArrayValue<NumericLikeType> ? NumArray : undefined;
+
+export function toNumArray(
+  arr: ArrayValue<NumericLikeType> | undefined,
+): NumArray | undefined {
+  if (!arr) {
+    return undefined;
+  }
+
+  if (isBigIntTypeArray(arr)) {
+    return Float64Array.from(arr, Number); // cast to float 64
```

Comment:
Data arrays that are passed to WebGL get converted to `Float32Array` soon after, unfortunately. I don't think it's worth optimising, though.